src/sidebar/services/groups.js

```diff
@@ -89,18 +95,26 @@ function groups(
     // link to an annotation in that group.
     const nonWorldGroups = groups.filter(g => g.id !== '__world__');
 
-    if (!directLinkedAnnotationId) {
+    if (!directLinkedAnnotationId && !directLinkedGroupId) {
       return Promise.resolve(nonWorldGroups);
     }
 
-    return api.annotation
-      .get({ id: directLinkedAnnotationId })
+    // If directLinkedGroup is the "Public" group, always return groups.
+    if (directLinkedGroupId === '__world__') {
+      return groups;
+    }
+
+    // If the directLinkedGroupId or the directLinkedAnnotationId's group
+    // is the "Public" group show the "Public" group otherwise do not.
```

Comment:
The first part of this comment is now redundant (`If the directLinkedGroupId`) given the part above.